vim/autoload/himalaya/shared/log.vim

```diff
@@ -7,4 +7,5 @@ function! himalaya#shared#log#err(msg)
   echohl ErrorMsg
   echomsg a:msg
   echohl None
+  throw a:msg
```

Comment:
There is one place in the code where this function is called many times in a row: https://github.com/soywod/himalaya/blob/14d8a3fecdf34aaf0d554ed08b60e48ed019e98b/vim/autoload/himalaya/shared/cli.vim#L14-L17

So I am not sure it's the best approach.

Reply:
I see! What would you think of extracting out the main body of `himalaya#msg#list` into a seperate function which does not catch the error and we can call from the preview and from `himalaya#msg#list` catching any error? I.e. something like (or maybe named differently if you don't want to rename the original function)
```vim
function! himalaya#msg#list()
  let mbox = himalaya#mbox#curr_mbox()
  let page = himalaya#mbox#curr_page()
  let msgs = s:cli("--mailbox %s list --page %d", [shellescape(mbox), page], printf("Fetching %s messages", mbox))
  let msgs = map(msgs, "s:format_msg_for_list(v:val)")
  let buftype = stridx(bufname("%"), "Himalaya messages") == 0 ? "file" : "edit"
  execute printf("silent! %s Himalaya messages [%s] [page %d]", buftype, mbox, page + 1)
  setlocal modifiable
  silent execute "%d"
  call append(0, s:render("list", msgs))
  silent execute "$d"
  setlocal filetype=himalaya-msg-list
  let &modified = 0
  execute 0
endfunction

function! himalaya#msg#try_list()
  try
    call himalaya#msg#list()
  catch
    if !empty(v:exception)
      redraw | call himalaya#shared#log#err(v:exception)
    endif
  endtry
endfunction
```

Reply:
Yes I like the idea, this way you can reuse the same function inside lua!